tests/controllers/UsersControllerTest.php

```diff
@@ -226,4 +226,192 @@ public function testThatUserDataIsNotDoubleEscapedOnUserEdit()
 
         $controller->updateUser($request, $db);
     }
+
+    /**
+     * Ensures that if the setTrusted method is called and no user_id is set,
+     * an exception is thrown
+     *
+     * @return void
+     *
+     * @test
+     * @expectedException        \Exception
+     * @expectedExceptionMessage You must be logged in to change a user account
+     */
+    public function testSetTrustedWithNoUserIdThrowsException()
+    {
+        $request = new \Request([], ['REQUEST_URI' => "http://api.dev.joind.in/v2.1/users/4/trusted", 'REQUEST_METHOD' => 'POST']);
+
+        $usersController = new \UsersController();
+        $db = $this->getMockBuilder('\JoindinTest\Inc\mockPDO')->getMock();
+
+        $usersController->setTrusted($request, $db);
+    }
+
+
+    /**
+     * Ensures that if the setTrsuted method is called and user_id is a,
+     * non-admin, an exception is thrown
+     *
+     * @return void
+     *
+     * @test
+     * @expectedException        \Exception
+     * @expectedExceptionMessage You must be an admin to change a user's trusted state
+     */
+    public function testSetTrustedWithNonAdminIdThrowsException()
+    {
+        $request = new \Request([], ['REQUEST_URI' => "http://api.dev.joind.in/v2.1/users/4/trusted", 'REQUEST_METHOD' => 'POST']);
+        $request->user_id = 2;
+        $usersController = new \UsersController();
+        // Please see below for explanation of why we're mocking a "mock" PDO
+        // class
+        $db = $this->getMockBuilder('\JoindinTest\Inc\mockPDO')->getMock();
+
+        $userMapper = $this->getMockBuilder('\UserMapper')
+            ->setConstructorArgs(array($db,$request))
+            ->getMock();
+
+        $userMapper
+            ->expects($this->once())
+            ->method('isSiteAdmin')
+            ->will($this->returnValue(false));
+
+        $usersController->setUserMapper($userMapper);
+        $usersController->setTrusted($request, $db);
+
+    }
+
+
+
+    /**
+     * Ensures that if the setTrusted method is called by an admin,
+     * but without a trusted state, an exception is thrown
+     *
+     * @return void
+     *
+     * @test
+     * @expectedException        \Exception
+     * @expectedExceptionMessage You must provide a trusted state
+     */
+    public function testSetTrustedWithoutStateThrowsException()
+    {
+        $request = $this->getMockBuilder('\Request')->disableOriginalConstructor()->getMock();
+        $request->method('getUserId')->willReturn(2);
```

Comment:
This is mocked to test/handle the getParameter call - in the other tests we never get to this point, so it's not needed